tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py

```diff
@@ -195,6 +195,13 @@ def process_request(self, request):
                 if old in body:
                     request.body = body.replace(old, new)
 
+            if request.body and request.uri and is_batch_payload(request):
+                import re
+                body = six.ensure_str(request.body)
+                matched_objects = set(re.findall(old, body))
```

Comment:
Use `set` to ensure that we don't iterate over the already replaced substring (no duplicates)